src/main/java/com/google/api/codegen/config/PackageMetadataConfig.java

```diff
@@ -0,0 +1,205 @@
+  }
+
+  /**
+   * The base name of the client library package. E.g., "google-cloud-logging-v1". Configured per
+   * language.
+   */
+  public String packageName(TargetLanguage language) {
+    return packageName().get(language);
+  }
+
+  /** A single-word short name of the API. E.g., "logging". */
+  public abstract String shortName();
+
+  /** The major version of the API, as used in the package name. E.g., "v1". */
+  public abstract String apiVersion();
+
+  /** The path to the API protos in the googleapis repo. */
+  public abstract String protoPath();
+
+  /** The author of the client library. */
+  public abstract String author();
+
+  /** The email of the author of the client library. */
+  public abstract String email();
+
+  /** The homepage of the client library. */
+  public abstract String homepage();
+
+  /** The name of the license of the client library. */
+  public abstract String licenseName();
+
+  private static Builder newBuilder() {
+    return new AutoValue_PackageMetadataConfig.Builder();
+  }
+
+  @AutoValue.Builder
+  protected abstract static class Builder {
+    abstract Builder gaxVersionBound(Map<TargetLanguage, VersionBound> val);
+
+    abstract Builder protoVersionBound(Map<TargetLanguage, VersionBound> val);
+
+    abstract Builder commonProtosVersionBound(Map<TargetLanguage, VersionBound> val);
+
+    abstract Builder packageName(Map<TargetLanguage, String> val);
+
+    abstract Builder packageVersionBound(Map<TargetLanguage, VersionBound> val);
+
+    abstract Builder shortName(String val);
+
+    abstract Builder apiVersion(String val);
+
+    abstract Builder protoPath(String val);
+
+    abstract Builder author(String val);
+
+    abstract Builder email(String val);
+
+    abstract Builder homepage(String val);
+
+    abstract Builder licenseName(String val);
+
+    abstract PackageMetadataConfig build();
+  }
+
+  /** Creates an PackageMetadataConfig with no content. Exposed for testing. */
+  @VisibleForTesting
+  public static PackageMetadataConfig createDummyPackageMetadataConfig() {
+    return newBuilder()
+        .gaxVersionBound(ImmutableMap.<TargetLanguage, VersionBound>of())
+        .protoVersionBound(ImmutableMap.<TargetLanguage, VersionBound>of())
+        .packageName(ImmutableMap.<TargetLanguage, String>of())
+        .commonProtosVersionBound(ImmutableMap.<TargetLanguage, VersionBound>of())
+        .packageVersionBound(ImmutableMap.<TargetLanguage, VersionBound>of())
+        .shortName("")
+        .apiVersion("")
+        .protoPath("")
+        .author("")
+        .email("")
+        .homepage("")
+        .licenseName("")
+        .build();
+  }
+
+  @SuppressWarnings("unchecked")
+  public static PackageMetadataConfig createFromFile(Path packageConfigPath) throws IOException {
+    Yaml yaml = new Yaml();
+    String contents = new String(Files.readAllBytes(packageConfigPath), StandardCharsets.UTF_8);
+    Map<String, Object> configMap = (Map<String, Object>) yaml.load(contents);
```

Comment:
I'm not sure see the benefit, other than for consistency. Creating a proto schema increases the number of places you need to touch if you want to change the schema, and it forces everyone to rebuild the generated Java classes. Given that the schema here is likely to be quite simple, do you think this is worthwhile?